src/components/filters/checkbox.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { FC, ReactNode } from "react"
+
+import { formatNumber } from "../../lib/formattingHelper"
+
+import Checkbox from "../../components/checkbox"
+
+interface Item {
+  value: string | number
+  label: string
+  renderIcon?: () => ReactNode
+}
+
+interface Props {
+  name: string
+  options: Item[]
+  applyFilters: (filtersToApply) => void
+  selected?: (string | number)[]
+  facet?: { [key: string]: number }
+}
+
+const CheckboxFilter: FC<Props> = ({
+  name,
+  options,
+  applyFilters,
+  selected = [],
+  facet = {},
+  children,
+}) => {
+  const values = options.map(({ value }) => value)
+
+  const selectValue = (value) => () => {
+    const selectionCopy = [...selected]
+
+    if (selectionCopy.includes(value)) {
+      const i = selectionCopy.indexOf(value)
+      selectionCopy.splice(i, 1)
+    } else {
+      selectionCopy.push(value)
+    }
+
+    selectionCopy.sort((a, b) =>
+      values.indexOf(a) < values.indexOf(b) ? -1 : 1
+    )
+
+    applyFilters({ [name]: selectionCopy })
+  }
+
+  const renderOption = ({ value, label, renderIcon }: Item) => {
+    const isSelected = selected.includes(value)
+    const resultsCount = facet[value.toString()]
+    const isDisabled = resultsCount === 0 && !isSelected
+
+    return (
+      <div className="w-12/12 -mb-5" key={`filter_${name}_${value}`}>
+        <Checkbox
+          name={`filter_${name}_${value}`}
+          renderLabel={() => (
+            <span className="text-base leading-xs flex items-center">
+              {renderIcon ? renderIcon() : null}
+              {label}
+              {typeof resultsCount === "number" && (
+                <span className="inline-block ml-5 text-grey-3 text-sm">
+                  ({formatNumber(resultsCount)})
```

Comment:
I would not expect the component to format the number for me, rather that renders what I pass...

Reply:
Fair, this allows us to not duplicate the formatting helper